File: linked-list/example-1.js
class ListNode {
	constructor(value, next = null) {
		this.next = next 
		this.value = value
	}
}

class SinglyLinkedList {
	constructor(head = null) {
		this.head = head
	}

	addFirst(value) {
		this.head = new ListNode(value, this.head)
	} // O(1)

	traverse(cb) {
		let pointer = this.head
		while (pointer) {
			cb(pointer.value)
			pointer = pointer.next
		}
	} // O(n)
}

const list = new SinglyLinkedList()

list.addFirst('groceries')
list.addFirst('movie')
list.addFirst('sfbc membership')

list.traverse(console.log)

File: linked-list/example-1/example-1.py
class ListNode:
	def __init__(self, value, next = None):
		self.value = value
		self.next = next

class SinglyLinkedList:
	def __init__(self, head = None):
		self.head = head

	def addFirst(self, value):
		self.head = ListNode(value, self.head)

	def traverse(self, cb):
		pointer = self.head
		while pointer:
			cb(pointer.value)
			pointer = pointer.next

list = SinglyLinkedList()

list.addFirst('groceries')
list.addFirst('movie')
list.addFirst('sfbc membership')

list.traverse(print)